File: Sample/java-sample/src/main/java/net/nend/customevent/mopub/BannerActivity.java
package net.nend.customevent.mopub;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.ViewGroup;

import com.mopub.mobileads.MoPubErrorCode;
import com.mopub.mobileads.MoPubView;

public class BannerActivity extends AppCompatActivity {

    private static final String MOPUB_AD_UNIT_ID = "YOUR_BANNER_UNIT_ID";
    private static final String TAG = "BannerActivity";

    private MoPubView mopubView;

    private MoPubView.BannerAdListener adListener = new MoPubView.BannerAdListener() {
        @Override
        public void onBannerLoaded(MoPubView banner) {
            Log.d(TAG, "onBannerLoaded:" + banner);
        }

        @Override
        public void onBannerFailed(MoPubView banner, MoPubErrorCode errorCode) {
            Log.d(TAG, "onBannerFailed:" + banner + "/ errorCode:" + errorCode);
        }

        @Override
        public void onBannerClicked(MoPubView banner) {
            Log.d(TAG, "onBannerClicked:" + banner);
        }

        @Override
        public void onBannerExpanded(MoPubView banner) {
            Log.d(TAG, "onBannerExpanded:" + banner);
        }

        @Override
        public void onBannerCollapsed(MoPubView banner) {
            Log.d(TAG, "onBannerCollapsed:" + banner);
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_banner);

        mopubView = findViewById(R.id.adview);
        mopubView.setAdUnitId(MOPUB_AD_UNIT_ID);
        mopubView.setBannerAdListener(adListener);
        mopubView.loadAd();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        mopubView.setBannerAdListener(null);
        mopubView.destroy();
        ViewGroup parent = (ViewGroup) mopubView.getParent();
        if (parent != null) {
            parent.removeAllViews();
        }
    }
}

File: Sample/kotlin-sample/src/main/java/net/nend/customevent/mopub/BannerActivity.kt
package net.nend.customevent.mopub

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.util.Log
import android.view.ViewGroup
import com.mopub.mobileads.MoPubErrorCode
import com.mopub.mobileads.MoPubView

class BannerActivity : AppCompatActivity() {

    private lateinit var mopubView: MoPubView

    private val adListener = object : MoPubView.BannerAdListener {
        override fun onBannerLoaded(banner: MoPubView) {
            Log.d(TAG, "onBannerLoaded:$banner")
        }

        override fun onBannerFailed(banner: MoPubView, errorCode: MoPubErrorCode) {
            Log.d(TAG, "onBannerFailed:$banner/ errorCode:$errorCode")
        }

        override fun onBannerClicked(banner: MoPubView) {
            Log.d(TAG, "onBannerClicked:$banner")
        }

        override fun onBannerExpanded(banner: MoPubView) {
            Log.d(TAG, "onBannerExpanded:$banner")
        }

        override fun onBannerCollapsed(banner: MoPubView) {
            Log.d(TAG, "onBannerCollapsed:$banner")
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_banner)

        mopubView = findViewById<MoPubView>(R.id.adview).apply {
            adUnitId = MOPUB_AD_UNIT_ID
            bannerAdListener = adListener
            loadAd()
        }
    }

    override fun onDestroy() {
        super.onDestroy()

        mopubView.bannerAdListener = null
        mopubView.destroy()
        val parent = mopubView.parent as ViewGroup
        parent.removeAllViews()
    }

    companion object {
        const val MOPUB_AD_UNIT_ID = "YOUR_BANNER_UNIT_ID"
        const val TAG = "BannerActivity"
    }
}
